### PR TITLE
refactor: Cookie and Set-Cookie header

### PR DESCRIPTION
**Pull request by nielsenko**
## Description
Hardens cookie header handling and reshapes the Set-Cookie types so the single-cookie value and the header collection are distinct, matching the request side.

- Rename the single-cookie class `SetCookieHeader` -> `SetCookie` (name/value + attributes)
- `SetCookieHeader` is now a `List<SetCookie>` collection, one element per `Set-Cookie` response line 
  (never comma-folded, RFC 6265 4.1).
- `SetCookieHeader` gains `.empty()`, `add`, `addAll`, value equality, and a multi-value codec
- `SetCookie.encode()` is now public.
- `CookieHeader.parse` is lenient: skip malformed cookies and keep the well-formed ones.
  One bad third-party cookie no longer hides a valid session/auth cookie. Throw only when no usable cookie remains. 
- `SetCookieHeader.parse` stays strict (server-produced, no skip rationale).
- Add `CookieHeader.getCookies(name)` returning every match in header order.
- `getCookie` returns the first of possible duplicates (RFC 6265 5.4).
- `SetCookie` strips a leading dot from the `Domain` (RFC 6265 5.2.3).

## Related Issues
- Related: #113 

## Pre-Launch Checklist
- [x] This update focuses on a single feature or bug fix.
- [x] I have read and followed the Dart Style Guide and formatted with dart format.
- [x] I have referenced at least one issue this PR fixes or is related to.
- [x] I have updated/added relevant documentation (doc comments with `///`).
- [x] I have added new tests to verify the changes.
- [x] All existing and new tests pass successfully.
- [x] I have documented any breaking changes below.

## Breaking Changes
- [x] Includes breaking changes.
- [ ] No breaking changes.

### CODE REVIEW

File: packages/relic/test/headers/typed/cookie_header_test.dart [hunk 1]
```diff
@@ -35,69 +35,83 @@ void main() {
...
-              (final e) => e.message,
-              'message',
-              contains('Invalid cookie format'),
-            ),
-          ),
+        final headers = await getServerRequestHeaders(
+          server: server,
+          touchHeaders: (final h) => h.cookie,
+          headers: {'cookie': 'sessionId=abc123; userId=42\x7F'},
         );
```
Comment (marcelomendoncasoares) [inline packages/relic/test/headers/typed/cookie_header_test.dart]: Suggestion: Also test invalid before valid to ensure skip is consistent regardless of order.

File: packages/relic_core/test/headers/typed/set_cookie_header_test.dart [hunk 2]
```diff
@@ -124,16 +136,64 @@ void main() {
...
+      });
+    });
+
+    group('Given an existing collection,', () {
+      test('when add is called, '
+          'then a new collection with the cookie appended is returned.', () {
+        final header = const SetCookieHeader.empty().add(a).add(b);
+
+        expect(header.cookies, equals([a, b]));
+      });
```
Comment (marcelomendoncasoares) [inline packages/relic_core/test/headers/typed/set_cookie_header_test.dart:336]: Question: Should we also test that `empty()` returns a `const` list that is not mutated from `add`/`addAll`?

File: packages/relic_core/lib/src/headers/typed/headers/set_cookie_header.dart [hunk 3]
```diff
@@ -242,9 +251,67 @@ final class SetCookieHeader {
...
+      value.cookies.map((final c) => c.encode()).toList();
+
+  @override
+  bool operator ==(final Object other) =>
+      identical(this, other) ||
+      other is SetCookieHeader &&
+          const ListEquality<SetCookie>().equals(cookies, other.cookies);
+
+  @override
+  int get hashCode => const ListEquality<SetCookie>().hash(cookies);
```
Comment (marcelomendoncasoares) [inline packages/relic_core/lib/src/headers/typed/headers/set_cookie_header.dart:318]: Question: Are we testing equality? Did not see on the tests added on this PR, but might exist on previously existing ones.

File: packages/relic_core/lib/src/headers/typed/headers/set_cookie_header.dart [hunk 4]
```diff
@@ -258,7 +325,7 @@ final class SameSite {
 
   /// Cookie with this value will be sent in all requests.
   ///
-  /// [Cookie.secure] must also be set to true, otherwise the `none` value
+  /// [SetCookie.secure] must also be set to true, otherwise the `none` value
   /// will have no effect.
   static const none = SameSite._('None');
```
Comment (marcelomendoncasoares) [inline packages/relic_core/lib/src/headers/typed/headers/set_cookie_header.dart:342]: Question: Should we have a test for `SetCookie` with `secure: false`?

File: packages/relic_core/lib/src/headers/typed/headers/cookie_header.dart [hunk 5]
```diff
@@ -28,24 +28,58 @@ final class CookieHeader {
...
+    // path-scoped duplicates).
+    //
+    // The server cannot tell them apart from the header alone. Keep such
+    // same-name cookies rather than rejecting the whole header.
+    // splitTrimAndFilterUnique still collapses byte-identical segments, which
+    // is harmless since they are indistinguishable. [getCookie] returns the
+    // first match.
+    //
+    // Cookie names are case-sensitive per RFC 6265 4.2.2/5.4, so `Sid` and
+    // `sid` stay distinct.
```
Comment (marcelomendoncasoares) [inline packages/relic_core/lib/src/headers/typed/headers/cookie_header.dart:48]: Question: Are we validating this on tests?

File: packages/relic/test/headers/typed/cookie_header_test.dart
Comment (marcelomendoncasoares) [inline packages/relic/test/headers/typed/cookie_header_test.dart:20]: Discussion: These tests are a bit out of the pattern because the group is rather generic and the test cases are described on the `when` clauses. Commenting here, but valid for the changes on the tests under this group. Since we're touching the cooking implementation, could be a good time to reorganize all at once. The correct would be the cookie description on the `Given` (i.e. "Given an empty cookie header, when passed to the server, then it responds...").

File: packages/relic_core/test/headers/typed/set_cookie_header_test.dart [hunk 6]
```diff
@@ -124,16 +136,64 @@ void main() {
...
+    group('Given a malformed Set-Cookie line,', () {
+      test('when decoded, '
+          'then it throws (strict decode)', () {
+        // Set-Cookie is server-produced, so there is no need to be lenient
+        expect(
+          () => SetCookieHeader.codec.decode(['a=1', 'no-equals', 'b=2']),
+          throwsFormatException,
+        );
+      });
+    });
```
Comment (marcelomendoncasoares) [inline packages/relic_core/test/headers/typed/set_cookie_header_test.dart:372]: Suggestion: Collapse these single-test groups into a `test` only, without the group.

File: packages/relic_core/test/headers/typed/set_cookie_header_test.dart [hunk 7]
```diff
@@ -124,16 +136,64 @@ void main() {
...
+        final header = const SetCookieHeader.empty().add(a).add(b);
+
+        expect(header.cookies, equals([a, b]));
+      });
+
+      test('when addAll is called, '
+          'then all cookies are appended in order.', () {
+        expect(SetCookieHeader([a]).addAll([b]).cookies, equals([a, b]));
+      });
+    });
```
Comment (marcelomendoncasoares) [inline packages/relic_core/test/headers/typed/set_cookie_header_test.dart:342]: Suggestion: Move the `existing collection` to the `setUp` of the "Given" group instead of declaring inside each test.

**Review by marcelomendoncasoares — CHANGES_REQUESTED** [review]
A few suggestions for coverage and improving the pattern on the tests.

**Review by marcelomendoncasoares — APPROVED** [review]
LGTM! 🚀
